provision-osx/tasks/install_xcode.yml

```diff
@@ -55,14 +62,40 @@
 # Check here for versions already installed
 -
   name: Install xcode versions
-  shell: "source ~/.bash_profile && xcversion install {{ item }}"
+  shell: "{{ xcversion_bin.stdout }} install {{ item }}"
   args:
     executable: /bin/bash
   with_items: "{{ xcode_versions }}"
   when: xcode_install_user is defined and xcode_install_password is defined
   register: xcode_install
   failed_when: xcode_cli_tools_install.rc != 0 and xcode_cli_tools_install.stdout.find('already installed') == -1
   changed_when: xcode_cli_tools_install.stdout.find('already installed') == -1
+  become: yes
   environment:
     XCODE_INSTALL_USER: "{{ xcode_install_user }}"
     XCODE_INSTALL_PASSWORD: "{{ xcode_install_password }}"
+
+-
+  name: Check xcode output
+  debug: msg="Checking {{ item.cmd }} output"
```

Comment:
So here we are checking all installs to ensure the credentials are valid is that correct? Won't the preceding task fail anyway if that is the case?